### ISSUE TITLE
Different results for `strval()` and `(string)`

### ISSUE THREAD

**Issue by BOPOH** [vimeo/psalm#10386]
As far as [I understood](https://www.php.net/manual/en/language.types.string.php#language.types.string.casting), `strval()` and `(string)` produce the same result, but psalm handles it differently: https://psalm.dev/r/deeb8284d9

**Comment by orklah** [comment]
Indeed, strval is documented as only returning string: https://github.com/vimeo/psalm/blob/d9d08f76f452fc8a8d4a50b176d3c9fc8232e42b/stubs/CoreGenericFunctions.phpstub#L1375

But it could be improved by making it a conditional return and allowing it to return numeric-string when the input is either an int, float or numeric-string
